src/main/scala/util/RecordMap.scala

```diff
@@ -0,0 +1,27 @@
+// See LICENSE.SiFive for license details.
+
+package freechips.rocketchip.util
+
+import Chisel._
+import chisel3.Record
+import scala.collection.immutable.ListMap
+
+final case class RecordMap[T <: Data](eltMap: ListMap[String, T])
```

Comment:
This should not be a `case class` because Chisel types do not conform to structural equality.

Reply:
removed the case class